yarn-project/aztec-rpc/src/simulator_oracle/index.ts

```diff
@@ -87,19 +86,17 @@ export class SimulatorOracle implements DBOracle {
     return await this.contractDataOracle.getPortalContractAddress(contractAddress);
   }
 
-  // TODO: currently stubbed will be implemented in: https://github.com/AztecProtocol/aztec-packages/issues/529
   /**
    * Retreives the L1ToL2Message associated with a specific message key
+   * Throws an error if the message key is not found
    *
    * @param msgKey - The key of the message to be retreived
    * @returns A promise that resolves to the message data, a sibling path and the
    *          index of the message in the the l1ToL2MessagesTree
    */
   async getL1ToL2Message(msgKey: Fr): Promise<MessageLoadOracleInputs> {
-    void msgKey; // this line can be removed its to appease the linter on this stub
-    const message = L1ToL2Message.empty().toFieldArray();
-    // TODO: note index will be requested from the database, stubbed as 0 for the meantime
-    const index = 0n;
+    const message = (await this.node.getL1ToL2Message(msgKey)).toFieldArray();
+    const index = (await this.node.findL1ToL2MessageIndex(msgKey.toBuffer()))!;
```

Comment:
this is an O(n) lookup each time, i think we should store the index with the message when the message is consumed. Ive made another ticket for that https://github.com/AztecProtocol/aztec-packages/issues/697

Reply:
ah good catch!